File: Leetle/tests/reference_solutions/python/binary_search.py
def search(nums, target):
    """
    Given an array of integers nums which is sorted in ascending order,
    and an integer target, write a function to search target in nums.
    If target exists, then return its index. Otherwise, return -1.
    """
    left, right = 0, len(nums) - 1
    while left <= right:
        mid = (left + right) // 2
        if nums[mid] == target:
            return mid
        elif nums[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1

# Main function to handle input from stdin
if __name__ == "__main__":
    import sys
    import ast
    input_data = sys.stdin.read().strip()
    lines = input_data.split('\n')

    # Parse array (first line)
    nums_str = lines[0].strip()
    nums = ast.literal_eval(nums_str)

    # Parse target (second line)
    target = int(lines[1].strip())

    result = search(nums, target)
    print(result)

File: Leetle/tests/reference_solutions/javascript/binary_search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
function search(nums, target) {
    let left = 0, right = nums.length - 1;
    while (left <= right) {
        const mid = Math.floor((left + right) / 2);
        if (nums[mid] === target) {
            return mid;
        } else if (nums[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    return -1;
}

// Main function to handle input from stdin
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').trim();
const lines = input.split('\n');

// Parse array (first line)
const nums = JSON.parse(lines[0].trim());

// Parse target (second line)
const target = parseInt(lines[1].trim());

const result = search(nums, target);
console.log(result);
